src/hooks/useCart/useCart.tsx

```diff
@@ -131,114 +115,19 @@ const mergeWithCart = (
     );
 };
 
-const filterQuotaWithAvailableProducts = (
-  quota: Quota,
-  products: CampaignPolicy[]
-): Quota => {
-  const filteredQuota: Quota = {
-    remainingQuota: []
-  };
-  transform(
-    quota.remainingQuota,
-    (result: Quota, itemQuota) => {
-      if (products.some(policy => policy.category === itemQuota.category))
-        result.remainingQuota.push(itemQuota);
-    },
-    filteredQuota
-  );
-
-  if (quota.globalQuota) {
-    filteredQuota.globalQuota = [];
-    transform(
-      quota.globalQuota!,
-      (result: Quota, itemQuota) => {
-        if (products.some(policy => policy.category === itemQuota.category))
-          result.globalQuota!.push(itemQuota);
-      },
-      filteredQuota
-    );
-  }
-
-  if (quota.localQuota) {
-    filteredQuota.localQuota = [];
-    transform(
-      quota.localQuota!,
-      (result: Quota, itemQuota) => {
-        if (products.some(policy => policy.category === itemQuota.category))
-          result.localQuota!.push(itemQuota);
-      },
-      filteredQuota
-    );
-  }
-
-  return filteredQuota;
-};
-
-const hasNoQuota = (quota: Quota): boolean =>
-  quota.remainingQuota.every(item => item.quantity === 0);
-
-const hasInvalidQuota = (quota: Quota): boolean =>
-  // Note: Invalid quota refers to negative quota received
-  quota.remainingQuota.some(item => item.quantity < 0);
-
 export const useCart = (
   ids: string[],
   authKey: string,
-  endpoint: string
+  endpoint: string,
+  quotaResponse?: Quota | null
```

Comment:
@sebastianquek please advise if this is okay 😅

Reply:
Perhaps we can think of what the cart needs to know? I noticed `Quota` now contains different types of quota, does the cart need to know about all the different types?

Also, when will this be `null` and not `undefined`?

Reply:
quotaResponse might be null because in the useQuota hook, we initialize the quotaResponse state to be null. Upon rendering, the useQuota hook should fetch the quota and replace the quotaResponse with an actual value. However, in the event that an error occurs, the quotaResponse will still remain null.

Reply:
This no longer exists in the code, let me know if this is still an issue.